gph/src/ripser.h

```diff
@@ -1014,7 +1015,8 @@ class ripser
                         }
                     }
                 }
-            } else if (get_index(get_zero_apparent_cofacet(e, 1)) == -1)
+            } else if (prepare_higher_dims &&
```

Comment:
Is this another case of branch prediction implying no overhead from doing this every time we see an edge rather than deciding at the beginning?

Reply:
We help the branch predictor. I don't know for sure if this would results in a speed-up.
And also, in my opinion, it is much clear why we add this check.

Reply:
Ok great!  It's only dimension 0 anyway, so not exactly where we have to optimize to death.